### PR TITLE
Overview of pan compiler development

### PR DESCRIPTION
**Pull request by iliclaey**

### CODE REVIEW

File: _posts/2016-06-20-panc-overview.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+`mvn -pl panc -Dtest=Test#method test`
+
+##### Run multiple methods in a specific class
+`mvn -pl panc -Dtest=Test#method1+method2 test`
+
+##### Running all clojure tests
+`mvn -pl panc clojure:test`
+
+### Running the project
+After building the project, al created files will be located in the *panc/target/* folder. The built project can be executed on the command line as follows:
```
Comment (jrha) [inline _posts/2016-06-20-panc-overview.md]: al**l**

File: _posts/2016-06-20-panc-overview.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#### Valid2 phase
+Checks the object dependencies and compiles them.
+
+#### Output
+The actual output is generated for every format that is passed as an argument. Every format has a corresponding **Formatter** class.
+
+#### SELF
+SELF is initially parsed as a variable and is then converted into the corresponding operator depending on what type SELF is (SelfSimpleVariable, SelfSimpleListVariable...). This is done in **PanParserAstUtils.astToVariable()**. Suppose we have the following statement
+
+<center> `'/x' = list(1); '/x' = prepend(SELF, 2)`.
```
Comment (jrha) [inline _posts/2016-06-20-panc-overview.md]: stray `<center>`?

Comment (DChambersSTFC) [inline _posts/2016-06-20-panc-overview.md]: If you wanted it to be centered, you could do
'<p align="center">'/x' = list(1); '/x' = prepend(SELF, 2)</p>'
which would look like
<p align="center">'/x' = list(1); '/x' = prepend(SELF, 2)</p>

File: _posts/2016-06-20-panc-overview.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: add a section like th eother posts, like
```
 ---
layout: article
title: Overview of the pan compiler
author: Isaura Claeys
category: documentation
---
```

File: _posts/2016-06-20-panc-overview.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+## Overview
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: shift all sections with one `#`?

File: _posts/2016-06-20-panc-overview.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+The clojure file *pan_compiler.clj* is the main entry point of the program. It takes care of the command line arguments, creates a new instance of the Compiler class and starts the compilation process. The program consists of different stages: compile, build, valid1, valid2 and output. All stages, except output, make use of a cache. For every stage a single threadPoolExecutor is created to which the different tasks are submitted. Submitting a task to a certain threadpool is done via the compiler.
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: and use max 100 chars per line.

File: _posts/2016-06-20-panc-overview.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+layout: article
+title: Overview of the pan compiler
+author: Isaura Claeys
+category: documentation
+---
+
+# Overview
+The clojure file *pan_compiler.clj* is the main entry point of the program. It takes care of the
+command line arguments, creates a new instance of the Compiler class and starts the compilation
+process. The program consists of different stages: compile, build, valid1, valid2 and output. All
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: as a general remark: all specific terminology is best put between backticks (or `*` as you did with the `pan_compiler.clj`) in this case the names of the stages, but also below some function/class names etc.

File: _posts/2016-06-20-panc-overview.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+Checks if there are undefined elements and if the type bindings are correct.
+
+### Valid2 phase
+Checks the object dependencies and compiles them.
+
+### Output
+The actual output is generated for every format that is passed as an argument. Every format has a
+corresponding **Formatter** class.
+
+### SELF
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: this is not a pahse, so maybe needs a new `## ` section

File: _posts/2016-06-20-panc-overview.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+**getInstance** method, which should perform the actual creation of a new object after doing some
+checks on the input of the function.
+
+## Adding a new built-in data type
+### /src/main/java/org/quattor/pan/dml/data
+The implementation of the new type should be placed in this folder. Depending on what type your
+going to add, you should inherit from the right class.
+
+### /src/main/java/org/quattor/pan/type/BaseType.java
+THe type should be added to the hashmap.
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: `THe`-> `The`

**Comment by stdweird** [comment]
retest this please

File: _posts/2016-06-20-panc-overview.md [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+
+#### Formatter
+You can specify a specific formatter to use for the compilation. For example: `@format=pan`.
+
+### Documentation
+The javadoc is automatically generated by maven when building the project. The online documentation
+of the built-in functions is kept separate. All online documentation is located in the
+*pan/panc-docs/* folder. To add information about a new function, add it to the
+*panc-docs/source/standard-functions/standard-functions.rst* file, which is written in
+reStructuredTest. This documentation should contain the name of the function, a synopsis and a description. Mind that all functions are sorted alphabetically.
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: `reStructuredTest` -> `reStructuredText`
also, add a newline before the `Mind`, line is a bit long

File: _posts/2016-06-20-panc-overview.md [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+type mytype = {
+    'entry1': string
+    'entry2': double(1..)
+}
+```
+
+##### LinkType
+A *LinkType* specifies the type of the link it refers to.
+For example: `type mylink = long(0..)*`. The value of this link will be a path that refers to a
+value, which needs to be a long >= 0. A *LinkType* cannot exist on its own, it needs to be
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: remove the `(0..) in the type and the `>= 0` here. it's not relevant to the link

File: _posts/2016-06-20-panc-overview.md [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+file: <p align="center">`@expect=org.quattor.pan.exception.SyntaxException regex`, </p>
+where regex is optional. If you expect a specific result, you should use an XPath. For example:  
+<p align="center">`@expect="/profile/result=1"`.</p>
+
+#### Expected dependencies
+You can declare what dependencies you expect from the source file. This should be declared as
+follows: <p align="center">`dep: template_name`. </p>
+
+#### Formatter
+You can specify a specific formatter to use for the compilation. For example: `@format=pan`.
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: why is this usefull?

File: _posts/2016-06-20-panc-overview.md [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+#### Run a single test class
+`mvn -pl panc -Dtest=Test test` where Test is the name of the test you want to run.
+
+#### Run multiple test classes
+`mvn -pl panc -Dtest=Test1,Test2 test`
+
+#### Run a single method in a specific class
+`mvn -pl panc -Dtest=Test#method test`
+
+#### Run multiple methods in a specific class
```
Comment (stdweird) [inline _posts/2016-06-20-panc-overview.md]: how whould you run the pan tests? and is it possible to run only part of the pan tests?

File: _posts/2016-06-20-panc-overview.md (hunk 9 above)
Comment (jrha) [inline _posts/2016-06-20-panc-overview.md]: Best practice for markdown is not to insert hard line-breaks unless completely necessary, as renderers interpret them differently (some ignore, some insert a `<br>`, some end the paragraph).

**Comment by ned21** [comment]
@jrha - any issues with merging this?

**Pull request merged**